### ISSUE TITLE
[🐞] OnError callback is being called after Checkout.start even though the payment works

### ISSUE THREAD

**Issue by jan-bednar** [paypal/paypalcheckout-ios#89]
### 🐞 Describe the Bug
When calling `PayPalCheckout.Checkout.start(presentingViewController:, onApprove:, onCancel:, onError)` `onError` block is called almost immediately with following error:

```
▿ OrderActionError
  ▿ errorGeneratingLSAT : RequestError
    ▿ decodingFailure : DecodingError
      ▿ valueNotFound : 2 elements
        - .0 : Swift.Bool
        ▿ .1 : Context
          ▿ codingPath : 1 element
            - 0 : CodingKeys(stringValue: "upgradeLowScopeAccessToken", intValue: nil)
          - debugDescription : "Expected Bool value but found null instead."
          - underlyingError : nil
```

But - payment can be made. This violates the documentation that states that:

```
Called when an error cannot be recovered from, and the transaction has been cancelled
```

### 🔬 Minimal Reproduction
call `PayPalCheckout.Checkout.start(presentingViewController:, onApprove:, onCancel:, onError)` and observe the onError callback

### 😕 Actual Behavior
`onError` block is called even though PayPalCheckout is still show on the screen and payment flows can be successfully finished.

### 🤔 Expected Behavior
`onError` block should not be called when then checkout process in not closed and payment flow continues

### 🌍 Environment

- iOS version: - 15.2
- SDK version (`Checkout.version`): - 0.79, 0.81

### ➕ Additional Context

onError callback is invoked:
<img width="1252" alt="Screenshot 2022-01-24 at 15 38 33" src="https://user-images.githubusercontent.com/39833836/150803306-118c9643-8528-47ad-b88d-08916a811527.png">

but Checkout continues to work:
![Simulator Screen Shot - iPhone 11 - 2022-01-24 at 15 38 38](https://user-images.githubusercontent.com/39833836/150803432-4611bfcd-799b-45f5-8075-890d8701f18b.png)

**Comment by jonathajones** [comment]
Hi @jan-bednar - 
Thanks for reporting the issue. It seems like there's a request failure at the top of the SDK lifecycle, which is causing the end user to fall back to the web flow, but also the error to be propagated back through the `onError` closure. You're right that this goes counter to our documentation - we'll investigate. 

We'll follow up here if we need more information.

**Comment by jonathajones** [comment]
Hi @jan-bednar  - 
The specific issue you encountered here (regarding the LSAT upgrade failure) has been resolved in the latest SDK release (`0.109.0`), and you will no longer get `onError` callbacks invoked due to that flow. Apologies for the delay here.
